explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+(dnssec)=
+# DNS Security Extensions (DNSSEC)
+
+DNS is a mapping system between names and IP addresses. Besides this main purpose that allows us to use friendly names instead of a sequence of numbers to reach a web site, it may store more information about a particular domain, such as:
+ * point of contact
+ * when it was last updated
+ * what are the authoritative name servers for the domain
+ * what are the mail exchangers for the domain (i.e., which systems are responsible for email for this domain)
+ * mapping of host names to IP addresses, and vice-versa
+ * information about other services, such as kerberos realms, LDAP servers, etc (usually on internal domains only)
+ * and more
+
+When DNS was first conceived, security wasn't a top priority yet. At its origins, DNS is susceptible to multiple vulnerabilities, and has many weaknesses. Most of them are a consequence of spoofing: there is no guarantee that the reply you got to a DNS query a) was not tampered with; b) came from the true source.
+
+This is not news, and other mechanisms on top of DNS and around it are in place to counteract that weakness. For example, the famous *HTTPS* padlock that can be seen when accessing most websites nowadays, which uses the TLS protocol to both authenticate the website, and encrypt the connection. It doesn't prevent DNS spoofing, and your web browser might still be tricked into attempting a connection with a fraudulent website, but the moment the TLS certificate is inspected, a warning will be issued to the user. Depending on local policies, the connection might be even immediately blocked. Still, DNS spoofing is a real problem, and TLS itself is subject to other types of attacks.
+
+## What is it?
+
+DNSSEC stands for Domain Name System (DNS) Security Extensions (SEC). It's an extension to DNS that introduces digital signatures so that each DNS response can be verified for:
+
+ * integrity: The answer was not tampered with and did not change during transit.
+ * authenticity: The data came from the true source, and not another entity impersonating the source.
+
+It's important to note that DNSSEC, however, will NOT encrypt the data: it is still sent in the clear.
+
+DNSSEC is based on public key cryptography, meaning that every DNS zone has a public/private key pair. The private key is used to sign the zone's DNS records, and the corresponding public key can be used to verify those signatures. This public key is also published in the zone, and anyone fetching records can also fetch the public key to verify the signature of the data.
```

Comment:
"This public key is also published in the zone. Anyone querying the zone can also fetch the public key to verify the signature of the data."

Reply:
nice